packages/paste-core/components/progress-steps/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+import {render} from '@testing-library/react';
+import {CustomizationProvider} from '@twilio-paste/customization';
+
+import {
+  ProgressSteps,
+  ProgressStepSeparator,
+  ProgressStepComplete,
+  ProgressStepCurrent,
+  ProgressStepError,
+  ProgressStepIncomplete,
+} from '../src';
+import {Divs, Buttons, Anchors} from '../stories/horizontal.stories';
+
+describe('ProgressSteps', () => {
+  it('should render divs correctly', () => {
+    const {queryAllByText} = render(<Divs />);
+    const [completeIcon, completeLabel] = queryAllByText('Complete');
+    const [incompleteIcon, incompleteLabel] = queryAllByText('Incomplete');
+    const [errorIcon, errorLabel] = queryAllByText('Error');
+    const [currentIcon, currentLabel] = queryAllByText('Current');
+
+    expect(completeIcon).toBeDefined();
+    expect(completeLabel).toBeDefined();
+    expect(incompleteIcon).toBeDefined();
+    expect(incompleteLabel).toBeDefined();
+    expect(errorIcon).toBeDefined();
+    expect(errorLabel).toBeDefined();
+    expect(currentIcon).toBeDefined();
+    expect(currentLabel).toBeDefined();
+
+    expect(completeLabel.parentElement?.tagName).toEqual('DIV');
+  });
+
+  it('should render buttons correctly', () => {
+    const {getByText} = render(<Buttons />);
+    const signupLabel = getByText('Sign up').parentElement;
+
+    expect(signupLabel?.tagName).toEqual('BUTTON');
+    expect(signupLabel?.getAttribute('type')).toEqual('button');
+  });
+
+  it('should render anchors correctly', () => {
+    const {getByText} = render(<Anchors />);
+    const signupLabel = getByText('Sign up').parentElement;
+
+    expect(signupLabel?.tagName).toEqual('A');
+    expect(signupLabel?.getAttribute('href')).toEqual('#');
+  });
+
+  it('handles incorrect props correctly', () => {
+    // hide console errors from terminal when throwing expected errors
+    const spy = jest.spyOn(console, 'error');
+    spy.mockImplementation(() => {});
+
+    expect(() =>
+      render(
+        <ProgressStepIncomplete as="div" disabled>
+          Only buttons disabled
+        </ProgressStepIncomplete>
+      )
+    ).toThrow();
```

Comment:
question: this has only bitten us once in the 4 years we've been doing it, where a team wasn't type checking correctly and it threw a customer facing error in prod. Should be keep doing it?